### PR TITLE
Closes #523 | Add/Update Dataloader PFSA-ID

### LINKED ISSUE #523: Create dataset loader for PFSA-ID
Dataloader name: `pfsa_id/pfsa_id.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?pfsa_id

|     Dataset| pfsa_id |
|-------------|---|
| Description | PFSA-ID is an annotated corpus for Public Figure Statement Attribution in the Indonesian Language. The annotation using the multi-class named entity recognition with 11 labels: PERSON, ROLE, AFFILIATION, PERSONCOREF, CUE, CUECOREF, STATEMENT, ISSUE, EVENT, DATETIME, and LOCATION and using the BILOU scheme as the representation of tokens. |
| Subsets | pfsa_id, pfsa_id_med, pfsa_id_test |
| Languages | ind |
| Tasks | Named Entity Recognition, Statement Tagging |
| License | Creative Commons Attribution Non Commercial Share Alike 4.0 (cc-by-nc-sa-4.0) |
| Homepage | https://github.com/sigit-purnomo/pfsa-id-dl |
| HF URL | - |
| Paper URL | https://doi.org/10.1108/GKMC-04-2022-0091, https://doi.org/10.1016/j.knosys.2024.111558 |

### PR DESCRIPTION
**Pull request by patrickamadeus**
Closes #523 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [.] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### pfsa_id
image

### pfsa_id_med
image

### pfsa_id_test
image

### CODE REVIEW

**Review by SamuelCahyawijaya — APPROVED** [review]
LGTM!

**Review by MJonibek — CHANGES_REQUESTED** [review]
@patrickamadeus Thank you for working on this dataloader. Your dataloader produces only three samples per split (which means one sample per file), however, inside the file, there are multiple sentences. I think you need to separate each sentence into a single sample, rather than unite them into one. 

What do you think @SamuelCahyawijaya?

**Review by SamuelCahyawijaya — CHANGES_REQUESTED** [review]
Hello @patrickamadeus @MJonibek, sorry, I missed this one before and thank you for noticing this one.

Yes, @MJonibek is correct, an instance for tagging task is commonly splitted per sentence in the conll format. So, it we should yield the instance when there is an empty row instead of just skipping the empty line.

File: seacrowd/sea_datasets/pfsa_id/pfsa_id.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        return split_gen
+
+    def _generate_examples(self, filepath: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        for i, path in enumerate(filepath):
+            tokens, labels = [], []
+            lines = open(path, "r").readlines()
+
+            for line in lines:
+                if line == "\n":
```
[the remark this answers is not in the document]
Comment (patrickamadeus) [reply seacrowd/sea_datasets/pfsa_id/pfsa_id.py:247]: Thanks for the review @MJonibek @SamuelCahyawijaya ! Please check the latest commit for the revision, it should have been wrapped up!

**Review by MJonibek — APPROVED** [review]
@patrickamadeus Great work, LGTM. One small issue: please rerun make command and solve issues it shows (screen attached)

image

**Review by SamuelCahyawijaya — APPROVED** [review]
Thank you for the update, @patrickamadeus! LGTM!

**Pull request merged**